### DISCUSSION TITLE
Node js port is already in use

### DISCUSSION THREAD

**Discussion by msiend** [community/community#123221] [Other Feature Feedback, Questions, & Ideas]
### Getting a Error Node js port is already in use .

Question

### Body

When I start my `webserver`, node throws listen `EADDRINUSE:` address already in use `:::3000`, I tried to use netstat and search for 3000 port (or another, it happens with any port), but nothing found. Also no node processes in task manager, no webpage on `localhost:3000`. Also I tried to reload windows, but nothing changed

**Comment by msi-end** [comment]
You can search on how to kill that process.
**For Linux/Mac OS search (sudo) run this in the terminal:**
`lsof -i tcp:3000 `
and then when you get the PID (3864) just kill the processes by PID (process ID)
` kill -9 PID`

**On Windows:**
```
netstat -ano | findstr :3000
tskill typeyourPIDhere 
```
**Or use :**
` killall -9 node`

the above command can exit vs code ssh connection when you are developing remotely and kill all node processes that can lead to problems especially if you have some apps on production using node , there is a better way to do it by using netstat to get all node processes with the port they are using and then kill the only one you want by PID

` netstat -lntp | grep node`

you will get all node processes :
 tcp6  0      0 :::5744    :::*    LISTEN     3864/node

and then when you get the PID (3864) just kill the processes by PID
`   kill -9 PID`
